File: backend/app/schemas/simulations.py
from typing import Any

from pydantic import BaseModel


class SimulationFileResponse(BaseModel):
    id: str
    dataset_id: str
    bim_dataset_id: str | None
    filename: str
    format: str
    status: str | None
    created_at: str | None
    metadata: dict[str, Any] | None


class SimulationVariableResponse(BaseModel):
    id: str
    simulation_dataset_id: str
    bim_space_id: str | None
    variable_id: str
    variable_name: str
    unit: str | None
    frequency: str | None
    key: str | None
    created_at: str | None


class SimulationTimeseriesResponse(BaseModel):
    id: str
    simulation_dataset_id: str
    variable_id: str
    timestamp: str | None
    value: float
    created_at: str | None

File: frontend/src/types/api/simulations.ts
export type SimulationProcessingMetadata = {
  variable_count: number | null;
  timestep_count: number | null;
  skipped_values: number | null;
  processed_at: string | null;
};

export type SimulationProcessingSummary = {
  metadata: SimulationProcessingMetadata;
};

export type SimulationFileDto = {
  id: string;
  dataset_id: string;
  bim_dataset_id: string | null;
  filename: string;
  format: string;
  status: string | null;
  created_at: string | null;
  metadata: Record<string, unknown> | null;
};

export type SimulationVariableDto = {
  id: string;
  simulation_dataset_id: string;
  bim_space_id: string | null;
  variable_id: string;
  variable_name: string;
  unit: string | null;
  frequency: string | null;
  key: string | null;
  created_at: string | null;
};

export type SimulationTimeseriesPointDto = {
  id: string;
  simulation_dataset_id: string;
  variable_id: string;
  timestamp: string | null;
  value: number;
  created_at: string | null;
};
